PocketKit/Sources/PocketKit/PocketNotificationService.swift

```diff
@@ -0,0 +1,170 @@
+        center.setNotificationCategories(Braze.Notifications.categories)
+        center.delegate = self
+        // We also ask for provisional authotization which is in iOS 12 and above and allows us to use push notifications that only post to the notification center
+        // https://www.braze.com/docs/user_guide/message_building_by_channel/push/ios/notification_options/#provisional-push
+        // Developer Note: Since we are registering provisional push, the user will never see the pop up to allow push notifications.
+        // This can make it challenging to test push notification registration flows because you will not see push registration appear, so you may want to remove this during active development.
+        center.requestAuthorization(options: [.badge, .sound, .alert, .provisional]) { granted, error in
+            print("Notification authorization, granted: \(granted), error: \(String(describing: error))")
+        }
+    }
+    
+    /**
+     Proxy function to braze to register a device token
+     */
+    public func register(deviceToken: Data) {
+        //TODO: Also register the token with the Pocket Instant Sync endpoint
+        braze.notifications.register(deviceToken: deviceToken)
+    }
+    
+    /**
+     Proxy function to braze to handle background notifications
+     */
+    public func handleBackgroundNotifcation(didReceiveRemoteNotification userInfo: [AnyHashable : Any],
+                                            fetchCompletionHandler completionHandler: @escaping (UIBackgroundFetchResult) -> Void) {
+        
+        // Forward background notification to Braze.
+        if braze.notifications.handleBackgroundNotification(
+            userInfo: userInfo,
+            fetchCompletionHandler: completionHandler
+        ) {
+            // Braze handled the notification, nothing more to do.
+            return
+        }
+        
+        
+        // Braze did not handle this remote background notification.
+        // We can handle the notification yourself here.
+        //TODO: Handle the V3 Instant sync push notification.
+        
+        // Manually call the completion handler to let the system know
+        // that the background notification is processed.
+        completionHandler(.noData)
+    }
+    
+    /**
+     Proxy function to Braze to handle user facing notifications
+     */
+    public func didRecieveUserNotifcation( _ center: UNUserNotificationCenter,
+                                           didReceive response: UNNotificationResponse,
+                                           withCompletionHandler completionHandler: @escaping () -> Void) {
+        let _ = braze.notifications.handleUserNotification(response: response, withCompletionHandler: completionHandler)
+    }
+}
+
+/**
+ Extend the Notification service to support User Notifications
+ */
+extension PocketNotificationService : UNUserNotificationCenterDelegate {
+    
+    func userNotificationCenter(
```

Comment:
This is the style we're shooting for (note how the first parameter is on its own line)